api-server/api-server-common/src/storage/storage_api/mod.rs

```diff
@@ -136,6 +140,56 @@ impl Utxo {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct FungibleTokenData {
+    pub token_ticker: Vec<u8>,
+    pub number_of_decimals: u8,
+    pub metadata_uri: Vec<u8>,
+    pub circulating_supply: Amount,
+    pub total_supply: TokenTotalSupply,
+    pub is_locked: bool,
+    pub frozen: IsTokenFrozen,
+    pub authority: Destination,
+}
+
+impl FungibleTokenData {
+    pub fn mint_tokens(mut self, amount: Amount) -> Self {
+        self.circulating_supply = (self.circulating_supply + amount).expect("no overflow");
+        self
+    }
+
+    pub fn unmint_tokens(mut self, amount: Amount) -> Self {
+        self.circulating_supply = (self.circulating_supply - amount).expect("no underflow");
+        self
+    }
+
+    pub fn freeze(mut self, is_token_unfreezable: IsTokenUnfreezable) -> Self {
+        self.frozen = IsTokenFrozen::Yes(is_token_unfreezable);
+        self
+    }
+
+    pub fn unfreeze(mut self) -> Self {
+        self.frozen = IsTokenFrozen::No(IsTokenFreezable::Yes);
+        self
+    }
+
+    pub fn lock(mut self) -> Self {
+        self.is_locked = true;
+        self
+    }
+
+    pub fn change_authority(mut self, authority: Destination) -> Self {
+        self.authority = authority;
+        self
+    }
+}
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode)]
+pub enum CoinOrTokenId {
```

Comment:
I recently moved this struct to `common/primitives` so you can reuse it now

Reply:
I saw that, but was not sure if I should add encode/decode to it

Reply:
added encode/decode and reused the existing enum